src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -1001,30 +987,87 @@ private static PlatformAttributes CopyAllAttributes(PlatformAttributes copyTo, P
                     attributes = CopyAttributes(containerAttributes);
                 }
 
-                AddPlatformAttributes(symbol.GetAttributes(), ref attributes);
-
-                if (symbol is IMethodSymbol method && method.IsAccessorMethod())
-                {
-                    // Add attributes for the associated Property
-                    AddPlatformAttributes(method.AssociatedSymbol.GetAttributes(), ref attributes);
-                }
+                MergePlatformAttributes(symbol.GetAttributes(), ref attributes);
 
                 attributes = platformSpecificMembers.GetOrAdd(symbol, attributes);
             }
 
             return attributes != null;
 
-            static bool AddPlatformAttributes(ImmutableArray<AttributeData> immediateAttributes, [NotNullWhen(true)] ref SmallDictionary<string, PlatformAttributes>? attributes)
+            static void MergePlatformAttributes(ImmutableArray<AttributeData> immediateAttributes, [NotNullWhen(true)] ref SmallDictionary<string, PlatformAttributes>? parentAttributes)
             {
-                bool added = false;
+                SmallDictionary<string, PlatformAttributes>? childAttributes = null;
                 foreach (AttributeData attribute in immediateAttributes)
                 {
                     if (s_osPlatformAttributes.Contains(attribute.AttributeClass.Name))
                     {
-                        added |= TryAddValidAttribute(ref attributes, attribute);
+                        TryAddValidAttribute(ref childAttributes, attribute);
+                    }
+                }
+
+                if (childAttributes != null)
+                {
+                    if (parentAttributes != null && parentAttributes.Any())
+                    {
+                        foreach (var (platform, attributes) in parentAttributes)
+                        {
+                            if (DenyList(attributes) &&
+                                !parentAttributes.Any(ca => AllowList(ca.Value)))
+                            {
+                                // if all are deny list then we can add the child attributes
+                                foreach (var (name, childAttribute) in childAttributes)
+                                {
+                                    if (parentAttributes.TryGetValue(name, out var existing))
```

Comment:
Revert condition and return to reduce nesting.

Reply:
It is not returning here nor in the revert